### ISSUE TITLE
Mailing and Campaigns

### ISSUE THREAD

**Issue by nbuonin** [NYC-Camp/nyc-camp-2015#29]
**Issue**
The site needs to send emails to users when:
* User creates site account (default Drupal behavior)
* User registers for the Camp as a whole
  * Do we want an email on each event registration? I suspect not.
* User submits session
* Admin approves session submission
* Admin emails event registrants

Additionally we need to sign up users as MailChimp contacts through the site using the MailChimp module, giving users the chance to self select topics they’re interested in.

**Proposed Approach**
TBD

(below are my notes which are intended to start us off, an approach hasn't been settled on)
It’s proposed that we use an email service, like Mandrill, to send transactional types of messages, like: “Thanks for signing up”, “Thanks for the session proposal”, “Watch out, this session has been moved”

MailChimp can be used to send “strategic communication” type messages, like: “Come to the camp”, “Sign up for this summit”, “Participate in Coding for a Cause”.  We’d need to import existing site users into MC before launch to ensure that every site user is signed up.  (Could we use signup records from last years camp to form the needed segments?  For example, could we take all the devops attendees, and check off a devops field in their MC record.)

We can use the form generated by the MC module, and attach it to the site user sign-up form to subscribe those to MC.

**Comment by nbuonin** [comment]
**Summits**
Ensure that, on summit pages, anon site visitors have a way to sign up for emails about that summit.

**Comment by willykaram** [comment]
I suggest that we split this up into three separate issues for: (1) general automated emails that need to be sent from the site, such as when someone registers, (2) manual emails that need to be sent from the site to users, such as when a summit organizer needs to communicate with conference attendees interested in a summit or users that have registered to attend a summit and (3) campaign related emails that we'll be sending from MailChimp.

**Comment by nbuonin** [comment]
As a baseline, I set up Mandrill on my multi-site branch, which can be brought over.  Given that, it might make sense for No. 1 to be distributed among the functions that require it.  So for example, who ever sets up something that generates an email, set up the rule to send that email.

For No 2., some one who's interested in a summit may not be an authenticated user, so that person's info can be collected by Mail Chimp, reached via MC.  For some one who's signed up for a summit, we can send to that person from the site - using Views send perhaps?

I think No. 3 would be best handled right from Mail Chimp.  I think they have a clean, easy to use interface, and account access can be delegated to sub-users.

That's my two cents, what do others think?

**Comment by nbuonin** [comment]
I dug into which email signup form would be best.  For anonymous users I think it makes the most sense to have a single page (which we could link to from summit pages) with an embedded form that posts directly to MailChimp: http://dev-nyc-camp-2015.gotpantheon.com/email-signup-test

There's a few reasons why I think this makes sense.  First, the form that the MailChimp contrib module generates doesn't update current MailChimp subscribers.  For example, if you are already are a current subscriber, and you want to add the front-end summit as something you're interested in, the module generated form will let you do that, and returns a confirmation message to the page -- BUT it doesn't update your record in MailChimp.  When you use the MC embedded form to do this, MC will send you an email to update your preferences.  They do this so that only the owner of that email can update their preferences.

Another reason is that our signup form has 18 fields to cover all areas of potential interest, and that would be unwieldy to embed on every summit page.  We could manually modify the form to only expose the fields that we want -- like only the checkbox for a given summit on that summit page.  But, if the user is updating preferences, they'll still have to go through the process I described above.  Given all that, I think it makes sense to ask anonymous users to sign-up off of a single form.

The contrib module is excellent for authenticated users - it just works.  I set up the email form on the users profile page and that updates the users MC profile with no fuss, and no extra steps for the user.

Another approach is to ask people to create site accounts first so we don't have to deal with the distinction between anon and logged in users.  That seems less confusing from a users point of view and easier for us to ensure that it will work.

What does everyone think?

**Comment by nbuonin** [comment]
Items 1 and 3 from @willykaram last comment are done.  I think that item no 2:

> "manual emails that need to be sent from the site to users, such as when a summit organizer needs to communicate with conference attendees interested in a summit or users that have registered to attend a summit"   

is contingent on the final config of registration.

As for approach, I think that we'd use both Views Send for timely sends, as well as some rules to add that user to the appropriate MC list.
